src/app/destinyTrackerApi/d2-reviewDataCache.ts

```diff
@@ -31,6 +32,16 @@ class D2ReviewDataCache {
     return this._getMatchingItemByReviewKey(reviewKey);
   }
 
+  _replaceRatingData(oldRatingData: D2RatingData, newRatingData: D2RatingData) {
+    const index = this._itemStores.indexOf(oldRatingData);
+
+    if (index < 0) {
+      return;
+    }
+
+    this._itemStores[index] = newRatingData;
```

Comment:
This is still a mutating update. You need to create an entirely new array if you mean for them to be completely updated. You may want to switch the store to be keyed off of something else though.